Mathlib/FieldTheory/Finite/Basic.lean

```diff
@@ -201,6 +228,84 @@ theorem sum_pow_units [DecidableEq K] (i : ℕ) :
         apply forall_congr'; intro x; simp [Units.ext_iff]
 #align finite_field.sum_pow_units FiniteField.sum_pow_units
 
+/-- The sum of a nontrivial subgroup of the units of a field is zero. -/
+theorem sum_subgroup_units_zero_of_ne_bot {G : Subgroup Kˣ} [Fintype G] (hg : G ≠ ⊥) :
+    ∑ x : G, (x.val : K) = 0 := by
+  rw [Subgroup.ne_bot_iff_exists_ne_one] at hg
+  rcases hg with ⟨a, ha⟩
+  -- The action of a on G as an embedding
+  have hinj := mul_right_injective a
+  let a_mul_emb : G ↪ G := ⟨(a * ·), hinj⟩
+  -- let a_mul_emb : G ↪ G := by exact mulLeftEmbedding a -- TODO factor this? In library?
+  -- ... and leaves G unchanged
+  have h_unchanged : Finset.univ.map a_mul_emb = Finset.univ := by simp
+  -- Therefore the sum of x over a G is the sum of a x over G
+  have h_sum_map := Finset.univ.sum_map a_mul_emb fun x => ((x : Kˣ) : K)
+  -- ... and the former is the sum of x over G.
+  -- By algebraic manipulation, we have Σ G, x = ∑ G, a x = a ∑ G, x
+  rw [h_unchanged, Function.Embedding.coeFn_mk] at h_sum_map
+  simp_rw [Subgroup.coe_mul, Units.val_mul, ← Finset.mul_sum] at h_sum_map
+  -- thus one of (a - 1) or ∑ G, x is zero
+  have hzero : (((a : Kˣ) : K) - 1) = 0 ∨ ∑ x : ↥G, ((x : Kˣ) : K) = 0 := by
+    rw [←mul_eq_zero, sub_mul, ← h_sum_map, one_mul, sub_self]
+  rcases hzero with h | h
+  · -- If the former, we reach contradiction from a ≠ 1
+    exfalso
+    apply ha
+    ext
+    rw [←sub_eq_zero]
+    exact h
+  · -- If the latter, we are done
+    exact h
+
+/-- The sum of a subgroup of the units of a field is 1 if the subgroup is trivial and 1 otherwise -/
+theorem sum_subgroup_units {G : Subgroup Kˣ} [Fintype G] [Decidable (G = ⊥)] :
+    ∑ x : G, (x.val : K) = if G = ⊥ then 1 else 0 := by
+  by_cases G_bot : G = ⊥
+  · subst G_bot
+    simp only [ite_true, Subgroup.mem_bot, Fintype.card_ofSubsingleton, Nat.cast_ite, Nat.cast_one,
+      Nat.cast_zero, univ_unique, Set.default_coe_singleton, sum_singleton, Units.val_one]
+  · simp only [G_bot, ite_false]
+    exact sum_subgroup_units_zero_of_ne_bot K G_bot
+
+-- example [Group G] (a : G) : G ≃ G := by
+--   exact Equiv.inv (sorryAx (Type ?u.888298) true)
+
+theorem sum_subgroup_pow_eq_zero {F : Type} [Field F] [Fintype F]
```

Comment:
could you refactor the theorems such as `sum_pow_lt_card_sub_one` to now be acorollary of this? I'll have a look at this proof closer later!

Reply:
also I'd probably call it `sum_pow_subgroup_eq_zero`

Reply:
I tried this, but `exists_pow_ne_one_of_isCyclic` is supposed to take a Type, whereas `sum_pow_lt_card_sub_one` is polymorphic over universes for the Field, and that seemed to cause issues.

It's sort of tragic that this is the case, I shouldn't have to deal with this for meat and potatoes theorems about fields.

Reply:
Notice that `sum_subgroup_pow_eq_zero` specifies a type `F` for the field different from the polymorphic one that the rest of the lemmas in this section get. Should I move the lemma? Should I fix `exists_pow_ne_one_of_isCyclic` to be polymorphic? Should I fix `K` to be `Type`?

Reply:
In fact the problems are weirder, there is one K shadowing another.

Reply:
I feel this is a question for the zulip

Reply:
I think you should try to make everything universe polymorphic, if possible

Reply:
Ok, I have just gone and made exists_pow_ne_one_of_isCyclic polymorphic, feel free to golf the other proof if you'd like.